### PR TITLE
Remove the judgment of the localrun mode (#3447)

### PR DESCRIPTION
**Pull request by tuteng**
### Motivation

Admin client need to be built in both localrun mode and non-localrun mode.
Initialization Admin client is required in class CmdTopics.
example:
```
topics = admin.topics();
```
More information: https://github.com/apache/pulsar/issues/3447

### Modifications

Remove the judgement of localrun mode

### Verifying this change
Test ok

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java [hunk 1]
```diff
@@ -205,23 +205,17 @@ public static void main(String[] args) throws Exception {
         }
 
         ++cmdPos;
-        boolean isLocalRun = cmdPos < args.length && "localrun".equals(args[cmdPos].toLowerCase());
 
         Function<PulsarAdminBuilder, ? extends PulsarAdmin> adminFactory;
```
Comment (ambition119) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java]: Function<PulsarAdminBuilder, ? extends PulsarAdmin> adminFactory = = (adminBuilder) -> {
            try {
                return adminBuilder.build();
            } catch (Exception ex) {
                System.err.println(ex.getClass() + ": " + ex.getMessage());
                System.exit(1);
                return null;
            }
        };

Comment (ambition119) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java]: Function<PulsarAdminBuilder, ? extends PulsarAdmin> adminFactory  = (adminBuilder) -> {  };
Is this also possible?

**Comment by ambition119** [comment]
👍

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java (hunk 1 above)
Comment (tuteng) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java]: It seems impossible. When I test, report missing return value in lambda expression, Causing compilation errors

**Comment by sijie** [comment]
@tuteng : 

> Admin client need to be built in both localrun mode and non-localrun mode.

I am not sure if that's the case. localrun invokes a script `bin/function-localrunner` to fork a process and run the process. so I explicitly disable creating admin client when localrun is used to avoid constructing unnecessary connections. 

so I think the right fix is for CmdTopics to handle if admin is null https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java#L57

**Comment by tuteng** [comment]
https://github.com/apache/pulsar/blob/26c1f3d6e2728f0f9ba76c99060c7627003b4d07/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/PulsarAdminTool.java#L113
Is this a better place to handle admin is null? Initialize only some required classes. @sijie 
```
if (admin != null) {
    jcommander.addCommand(c.getKey(), c.getValue().getConstructor(PulsarAdmin.class).newInstance(admin));
} else if (c.getKey().equals("functions") || c.getKey().equals("source") || c.getKey().equals("sink")) {
    jcommander.addCommand(c.getKey(), c.getValue().getConstructor(PulsarAdmin.class).newInstance(admin));
}
```

**Comment by sijie** [comment]
@tuteng your last comment looks good to me.

**Comment by tuteng** [comment]
run integration tests
run java8 tests

**Pull request merged**